Makefile

```diff
@@ -70,9 +71,16 @@ fix-lint:
 	golangci-lint run --fix
 
 .PHONY: verify-shellcheck
-verify-shellcheck:
+verify-shellcheck: install-shellcheck
 	PASSES="shellcheck" ./scripts/test.sh
 
+.PHONY: install-shellcheck
+install-shellcheck:
+	if ! command -v shellcheck > /dev/null 2>&1; then \
+        wget -qO- "https://github.com/koalaman/shellcheck/releases/download/${SHELLCHECK_VERSION}/shellcheck-${SHELLCHECK_VERSION}.linux.x86_64.tar.xz" | tar -xJv && \
+        cp "shellcheck-${SHELLCHECK_VERSION}/shellcheck" /usr/bin/; \
```

Comment:
copying to `/usr/bin/` is not a great idea as it requires root permissions.

Reply:
/usr/local/bin/ ? What do you think?  Maybe have better way @serathius

Reply:
Somehow it passed. No any error in the workflow. 

```
if ! command -v shellcheck > /dev/null 2>&1; then \
        wget -qO- "https://github.com/koalaman/shellcheck/releases/download/v0.8.0/shellcheck-v0.8.0.linux.x86_64.tar.xz" | tar -xJv && \
SUCCESS
        cp "shellcheck-v0.8.0/shellcheck" /usr/bin/; \
fi
PASSES="shellcheck" ./scripts/test.sh
Running with --race
Starting at: Wed Nov 30 21:41:14 UTC 2022

'shellcheck' started at Wed Nov 30 21:41:14 UTC 2022
% 'shellcheck' '-fgcc' 'scripts/build-binary.sh' 'scripts/build-docker.sh' 'scripts/build-release.sh' 'scripts/build.sh' 'scripts/build_lib.sh' 'scripts/build_tools.sh' 'scripts/codecov_upload.sh' 'scripts/fix.sh' 'scripts/fuzzing.sh' 'scripts/genproto.sh' 'scripts/install-marker.sh' 'scripts/measure-test-flakiness.sh' 'scripts/release.sh' 'scripts/release_mod.sh' 'scripts/test.sh' 'scripts/test_lib.sh' 'scripts/update_dep.sh' 'scripts/update_proto_annotations.sh' 'scripts/updatebom.sh' 'scripts/verify_genproto.sh' 'scripts/verify_proto_annotations.sh'
'shellcheck' completed at Wed Nov 30 21:41:19 UTC 2022
SUCCESS

```

But the better way seems to be something like below. Could you try this out? 
```
mkdir -p $HOME/.local/bin
cp "shellcheck-${SHELLCHECK_VERSION}/shellcheck" $HOME/.local/bin


```

Reply:
Two notes, it's bad to touch user home directory and `$HOME/.local/bin` is not guaranteed to be in `$PATH` environment variable especially when it doesn't exist. 

My recommendation is to put shellcheck in `./bin/tools` and don't assume that it is available in `PATH`

Reply:
FYI. https://github.com/actions/runner-images/pull/2491 

Another option is https://github.com/etcd-io/etcd/pull/14872#issuecomment-1333312588

Reply:
Runner != Developer